### PR TITLE
Add copysign to keras.ops.numpy

### LINKED ISSUE #23497: Add `keras.ops.copysign`
## Description

`keras.ops` covers the IEEE-754 sign and exponent family fairly completely, such as, `signbit`, `nextafter`, `ldexp`, `hypot`, `heaviside`, `logaddexp`, `logaddexp2` are all there, but `copysign` is missing.

It is available in every framework Keras builds on:

| | available |
|---|---|
| `numpy.copysign` | yes |
| `jax.numpy.copysign` | yes |
| `torch.copysign` | yes |
| `math.copysign` (Python stdlib) | yes |
| `keras.ops.copysign` | **no** |

It is also part of the [Python Array API standard](https://data-apis.org/array-api/latest/API_specification/generated/array_api.copysign.html) (added in the 2023.12 revision), which Keras generally tracks for `keras.ops.numpy`.

### Why it is worth having

`copysign(x1, x2)` returns the magnitude of `x1` with the sign of `x2`. The natural workaround in Keras today is

```python
keras.ops.abs(x1) * keras.ops.sign(x2)
```

which is silently wrong in two ways:

* it returns `0` wherever `x2 == 0`, instead of `±|x1|`;
* it cannot distinguish `-0.0` from `0.0`, so it never produces a negative result for a negative zero.

```python
>>> import numpy as np
>>> np.copysign(3.0, 0.0), np.copysign(3.0, -0.0)
(3.0, -3.0)
>>> abs(3.0) * np.sign(0.0), abs(3.0) * np.sign(-0.0)
(0.0, -0.0)
```

That correctness gap matters wherever sign is carried separately from magnitude including sign-magnitude and symmetric quantization, Householder reflections in QR, round-half-away-from-zero via `trunc(x + copysign(0.5, x))`, and sign-based gradient methods.

### Consistency across backends

Because there is no single op today, users hand-roll it, and the hand-rolled version behaves differently per backend around zero. A single `keras.ops.copysign` gives one definition on TensorFlow, JAX, NumPy, PyTorch and OpenVINO.

NumPy, JAX and PyTorch have a native `copysign` to delegate to. TensorFlow has no `tf.math.copysign`, but it can be composed exactly from the existing `signbit` in the TensorFlow backend. OpenVINO can be composed from `less`/`equal`/`select`/`negative` following the same sign-bit test its existing `signbit` uses.

### Proposal

Add `keras.ops.copysign` / `keras.ops.numpy.copysign`, implemented on all five backends, following `nextafter` for dtype promotion (`result_type(x1, x2, float)`), which matches `jax.numpy.copysign` across the whole dtype matrix.

I have this implemented and tested on all five backends and will open a PR shortly.

### PR DESCRIPTION
**Pull request by Neilblaze**
## Description

`keras.ops` already has most of the IEEE-754 sign and exponent family (`signbit`, `nextafter`, `ldexp`, `hypot`, `heaviside`, `logaddexp`, `logaddexp2`), but `copysign` was missing. It exists as `numpy.copysign`, `jax.numpy.copysign`, `torch.copysign` and `math.copysign`, and it is part of the [Python Array API standard](https://data-apis.org/array-api/latest/API_specification/generated/array_api.copysign.html) (2023.12).

Without it the usual workaround is `abs(x1) * sign(x2)`, which returns zero wherever `x2` is zero and can never produce a negative result from `-0.0`. Users end up hand-rolling it, and the hand-rolled version behaves differently per backend around zero. This adds one definition shared by all five.

NumPy, JAX and PyTorch delegate to their native `copysign`. TensorFlow has no `tf.math.copysign`, so it composes one from the `signbit` already in that backend. OpenVINO negates `x1` when the two sign bits differ, reusing the same `-0.0` test its own `signbit` uses.

Dtype promotion follows `nextafter`, the closest existing op: `result_type(x1, x2, float)`. I checked that against `jax.numpy.copysign` over all 121 ordered pairs of `ALL_DTYPES` and it agrees on every one, so the new dtype test compares against JAX the way the other two-input dtype tests do.

`_copysign` also provides a backend-agnostic fallback, following the shape `cbrt`, `hsplit` and `vsplit` use. When a backend does not implement `copysign`, it promotes both operands and negates `x1` wherever the two sign bits differ, which is the same shape the OpenVINO backend uses.

Fixes #23497

### Testing

Adds the four blocks a two-input op normally gets (dynamic shape, static shape, correctness, and the JAX-referenced dtype matrix), placed next to `nextafter`.

The correctness and dtype tests run under both settings of the backend-agnostic flag, so the fallback is covered too. `BACKEND_AGNOSTIC_OPS` moves to module scope because it was defined after `NumpyTwoInputOpsCorrectnessTest`, which is where the `copysign` correctness test lives.

`pytest keras/src/ops/numpy_test.py -k copysign` passes on all five backends, and the full `numpy_test.py` run shows no regressions on any of them.

I also compared each backend against `np.copysign` bit for bit over roughly 5.8k value pairs per dtype in float64, float32 and float16, including `±0.0`, `±inf`, `±NaN` and subnormals. Sign bits, NaN placement and values all match exactly.

> [!NOTE]
> One note on the API files. Only the four `copysign` export lines are included here. Running `api_gen.py` locally also rewrites about 130 unrelated files, but that same churn shows up on an unmodified `master`, so it is local tooling drift rather than something this PR introduces.

## Contributor Agreement

**Please review our [AI-Assisted Contribution Policy](../CONTRIBUTING.md#ai-assisted-contribution-policy) and check all boxes below before submitting your PR for review:**

- [x] I am a human, and not a bot.
- [x] I will be responsible for responding to review comments in a timely manner.
- [x] I will work with the maintainers to push this PR forward until submission.

**Note:** Failing to adhere to this agreement may result in your future PRs no longer being reviewed.

### CODE REVIEW

File: keras/src/ops/numpy.py [hunk 1]
```diff
@@ -2220,6 +2220,46 @@ def copy(x):
...
+
+    Example:
+    >>> x1 = keras.ops.convert_to_tensor([-1.0, 2.0, -3.0])
+    >>> x2 = keras.ops.convert_to_tensor([1.0, -1.0, -0.0])
+    >>> keras.ops.copysign(x1, x2)
+    array([ 1., -2., -3.], dtype=float32)
+    """
+    if any_symbolic_tensors((x1, x2)):
+        return Copysign().symbolic_call(x1, x2)
+    return backend.numpy.copysign(x1, x2)
```
Comment (hertschuh) [inline keras/src/ops/numpy.py]: Can you add a fallback implementation for other backends?

Here are some examples to show the pattern:
- https://github.com/keras-team/keras/pull/23570
- https://github.com/keras-team/keras/pull/23541

**Review by hertschuh — CHANGES_REQUESTED** [review]
Thanks for adding this!

File: keras/src/ops/numpy.py (hunk 1 above)
Comment (Neilblaze) [reply keras/src/ops/numpy.py]: @hertschuh Addressed. PTAL!

**Review by hertschuh — APPROVED** [review]
Thanks for adding this!
